dot/simswap/util/reverse2original.py

```diff
@@ -100,89 +106,103 @@ def reverse2wholeimage(
     oriimg,
     pasring_model=None,
     norm=None,
-    use_mask=False,
+    use_mask=True,
     use_gpu=True,
+    use_cam=True,
 ):
 
-    target_image_list = []
-    img_mask_list = []
+    device = torch.device("cuda" if use_gpu else "cpu")
     if use_mask:
-        if use_gpu:
-            smooth_mask = SoftErosion(
-                kernel_size=17, threshold=0.9, iterations=7
-            ).cuda()
-        else:
-            smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).cpu()
+        smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).to(
+            device
+        )
     else:
         pass
 
+    img = K.utils.image_to_tensor(oriimg).float().to(device)
+    img /= 255.0
+
     for swaped_img, mat, source_img in zip(swaped_imgs, mats, b_align_crop_tenor_list):
-        swaped_img = swaped_img.cpu().detach().numpy().transpose((1, 2, 0))
-        img_white = np.full((crop_size, crop_size), 255, dtype=float)
-
-        # inverse the Affine transformation matrix
-        mat_rev = np.zeros([2, 3])
-        div1 = mat[0][0] * mat[1][1] - mat[0][1] * mat[1][0]
-        mat_rev[0][0] = mat[1][1] / div1
-        mat_rev[0][1] = -mat[0][1] / div1
-        mat_rev[0][2] = -(mat[0][2] * mat[1][1] - mat[0][1] * mat[1][2]) / div1
-        div2 = mat[0][1] * mat[1][0] - mat[0][0] * mat[1][1]
-        mat_rev[1][0] = mat[1][0] / div2
-        mat_rev[1][1] = -mat[0][0] / div2
-        mat_rev[1][2] = -(mat[0][2] * mat[1][0] - mat[0][0] * mat[1][2]) / div2
-
-        orisize = (oriimg.shape[1], oriimg.shape[0])
+
+        img_white = torch.full((1, 3, crop_size, crop_size), 1.0, dtype=torch.float).to(
+            device
+        )
+
+        # invert the Affine transformation matrix
+        mat_rev = torch.ones([3, 3]).to(device)
```

Comment:
this operation does not depend on any data and must be moved outside of the loop

Reply:
same for any other in the loop, I think there is more

Reply:
I think I removed them all from the loop https://github.com/sensity-ai/dot/pull/20/commits/40051c2389b20d24592099882ebe80fa808a7428